deker_server_adapters/base.py

```diff
@@ -69,12 +67,17 @@ def hash_ring(self) -> HashRing:
         return hash_ring  # type: ignore[attr-defined]
 
     @property
-    def nodes(self) -> List[str]:
+    def nodes(self) -> List[Node]:
         """Return list of nodes."""
-        nodes = self.ctx.extra["nodes"]  # type: ignore[attr-defined]
-        if not nodes:
-            raise AttributeError("Attempt to use cluster logic in single server mode")
-        return nodes
+        config: ClusterConfig = self.ctx.extra.get("cluster_config")  # type: ignore[attr-defined]
+        if not config:
+            return []
+        return config.current
```

Comment:
I believe `self.nodes` shall always return a list of >=1 node.
If server is not in the cluster mode, it shall return just 1 address in the list. Otherwise it shall return all the cluster nodes' addresses. In both cases `self.client.base_url` shall be in the list.